### PR TITLE
Adding a workflow

### LINKED ISSUE #6: Feature: Add GitHub workflow
**Acceptance Criteria:**
-  This is work in progress

### PR DESCRIPTION
**Pull request by ahmdhjj**

### CODE REVIEW

File: .github/workflows/dotnet-desktop.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.0
+
+    - name: Setup MSBuild.exe
+      uses: microsoft/setup-msbuild@v2
+
+    - name: Install ntools and dev tools
```
Comment (naz-hage) [inline .github/workflows/dotnet-desktop.yml]: @ahmdhjj No need to insyall all devtools.  The runner already has git and VS 2022. We just need to install ntools via PowerShell the same way we did in DevSetup.ps1

**Review by naz-hage — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet-desktop.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      working-directory: ${{ github.workspace }}
+   
+    - name: Configure Git
+      run: |
+        git config --global user.name "gitHub-actions"
+        git config --global user.email "actions@github.com"
+
+    - name: Build using ntools
+      run: |
+        & "$env:ProgramFilesPath/nbuild/nb.exe" solution
```
Comment (naz-hage) [inline .github/workflows/dotnet-desktop.yml]: @ahmdhjj I recommend that we can combine `solution` and `test` target and use `staging` target,

File: .github/workflows/dotnet-desktop.yml (hunk 1 above)
Comment (ahmdhjj) [reply .github/workflows/dotnet-desktop.yml]: @naz-hage done

File: .github/workflows/dotnet-desktop.yml (hunk 2 above)
Comment (ahmdhjj) [reply .github/workflows/dotnet-desktop.yml]: @naz-hage done

**Review by naz-hage — APPROVED** [review]

**Pull request merged**